### PR TITLE
Expose methods on container models

### PR DESCRIPTION
**Pull request by kevensen**
Depends On:
ManageIQ/manageiq-providers-openshift#2
ManageIQ/manageiq#14798

https://bugzilla.redhat.com/show_bug.cgi?id=1440141

### CODE REVIEW

**Comment by gmcculloug** [comment]
@simon3z Can you please review this from the container side.  At a quick glance I'm a little concerned that we are exposing too much here.

**Comment by kevensen** [comment]
@gmcculloug I've updated the [bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1440141) to better describe what I'm hoping to accomplish.  I am absolutely open to any recommendations for improvement.

File: lib/miq_automation_engine/service_models/miq_ae_service_container_route.rb [hunk 1]
```diff
@@ -8,5 +8,9 @@ class MiqAeServiceContainerRoute < MiqAeServiceModelBase
     expose :labels,                   :association => true
     expose :is_tagged_with?
     expose :tags
+    expose :spec
```
Comment (mkanoor) [inline lib/miq_automation_engine/service_models/miq_ae_service_container_route.rb]: @kevensen 
Can the spec, spec=,annotations, tidy_provider_definition be moved into a mixin and then we can include it in 3 files

Comment (kevensen) [inline lib/miq_automation_engine/service_models/miq_ae_service_container_route.rb]: Yes they can.  I'll do that.

**Comment by simon3z** [comment]
@kevensen can you provide the other PR dependencies for this one? I guess we'll need the backend for these before and there may be changes (e.g. IIRC we're planning to rename `add_role_to_user`, etc.)

**Comment by kevensen** [comment]
**Depends On:**
https://github.com/ManageIQ/manageiq-providers-openshift/pull/2
https://github.com/ManageIQ/manageiq/pull/14798

**Comment by gmcculloug** [comment]
Moved dependences from comment above (https://github.com/ManageIQ/manageiq-automation_engine/pull/7#issuecomment-302399718) to the PR description so they are easy to identify.

**Comment by gmcculloug** [comment]
@kevensen Looks like the dependent PRs are stale.  Is this still being worked on or should we close for now and re-open if there is need?

**Comment by kevensen** [comment]
@gmcculloug Let’s go ahead and close this and the related PR’s for now.

**Pull request closed without merging**